python/src/aiconfig/editor/client/src/components/prompt/model_settings/ModelSettingsRenderer.tsx

```diff
@@ -22,6 +23,35 @@ const useStyles = createStyles(() => ({
   },
 }));
 
+type ErrorFallbackProps = {
+  settings?: JSONObject;
+  toggleJSONEditor: () => void;
+};
+
+function SettingsErrorFallback({
+  settings,
+  toggleJSONEditor,
+}: ErrorFallbackProps) {
+  const { resetBoundary: clearRenderError } = useErrorBoundary();
+  return (
+    <Flex direction="column">
+      <Text color="red" size="sm">
+        <Flex justify="flex-end">
+          <JSONEditorToggleButton
+            isRawJSON={false}
+            setIsRawJSON={() => {
+              clearRenderError();
+              toggleJSONEditor();
+            }}
+          />
+        </Flex>
+        Invalid settings format for model. Toggle JSON editor to update
+      </Text>
+      <JSONRenderer content={settings} />
+    </Flex>
+  );
```

Comment:
nit: can also make a follow up issue to centralize this with the similar Fallback function from prompt input but change the text. Not a blocker

Reply:
Main difference is input has the JSON toggle on the bottom right and settings has it top-right; IMO, refactor to support dynamic toggle placement and callbacks for this 14-line component would just be overkill